### PR TITLE
[2.8.x] Run scripted tests on sbt 1.4.x + refactoring

### PR DESCRIPTION
**Pull request by mkurz**
Backport of #10655
Only difference: The 2.8.x branch still runs tests against sbt 0.13.x

Requires #10712 merged to make the build green for this PR.

Also, like mentioned in #10655 I am aware that the cron conditions are not here yet. I will add them after #10712 got merged. Then I will restart Travis for this pull request here so all sbt versions will be tested with the fixes in #10712, also see see https://github.com/playframework/playframework/pull/10655#discussion_r583106363 Afterwards I can remove undesired sbt versions and add the cron config.

### CODE REVIEW

**Comment by mkurz** [comment]
@ignasi35 @octonato This backport ready as well!

Example how a cron build will look like (except without sbt 1.2):
https://travis-ci.com/github/playframework/playframework/builds/219526837
which is the build of this commit:
image
(the commit before I added the cron conditions)

File: .travis.yml [hunk 1]
```diff
@@ -55,76 +65,98 @@ jobs:
...
-      name: "Run tests for sbt 1.3.x"
-      script: scripts/test-scripted-13x
+    # Test against sbt 0.13.x and Scala 2.13.x, but only for cron builds
+    # (sbt 0.13.x / Scala 2.12.x was tested above already)
+    - stage: cron-test-sbt-0.13.x
+      name: "Run tests for sbt 0.13.x and Scala 2.13.x"
+      script: scripts/test-scripted $SCRIPTED_SBT_0_13 $TEST_SCALA_2_13
+      if: type = cron
+      workspaces:
+        use: published-local
```
Comment (ignasi35) [inline .travis.yml:142]: I wouldn't even care about 0.13 2.13

File: .travis.yml [hunk 2]
```diff
@@ -55,76 +65,98 @@ jobs:
...
-    - name: "Run scripted tests (b) for sbt 0.13.x"
-      script: scripts/test-scripted 0.13.x 'play-sbt-plugin/*2of3'
+    - name: "Run scripted tests (b) for sbt 0.13.x and Scala 2.12.x"
+      script: scripts/test-scripted $SCRIPTED_SBT_0_13 $TEST_SCALA_2_12 'play-sbt-plugin/*2of3'
       workspaces:
         use: published-local
-    - name: "Run scripted tests (c) for sbt 0.13.x"
-      script: scripts/test-scripted 0.13.x 'play-sbt-plugin/*3of3'
+    - name: "Run scripted tests (c) for sbt 0.13.x and Scala 2.12.x"
+      script: scripts/test-scripted $SCRIPTED_SBT_0_13 $TEST_SCALA_2_12 'play-sbt-plugin/*3of3'
```
Comment (ignasi35) [inline .travis.yml:91]: I'm about to suggest we move anything 0.13 to CRON jobs and let PR builds run lighter.

And, by all means, test only 0.13 scala 2.12: late laggards.

**Review by ignasi35 — CHANGES_REQUESTED** [review]
Let's remove a few jobs from the build. The matrix has too many axis and 0.13 should note taking so much attention. Also tempted to remove 1.4x2.12 since early adopters of sbt are likely to be early adopters of scala.

**Comment by ignasi35** [comment]
Do we need any mention of 1.4.x support in docs or migration guide?

**Comment by mkurz** [comment]
@ignasi35 Done, please see the commits I pushed.

> Do we need any mention of 1.4.x support in docs or migration guide?

Not really, there is no action a user needs to take. Just upgrade to Play 2.8.8 and everything works. We can mention that Play now fully supports sbt 1.4 on the [GitHub release page](https://github.com/playframework/playframework/releases) however.

**Review by ignasi35 — APPROVED** [review]

**Pull request merged**